File: test/test.partial.js
/* global describe, it, require */
'use strict';

// MODULES //

var // Expectation library:
	chai = require( 'chai' ),

	// Module to be tested:
	partial = require( './../lib/partial.js' );


// VARIABLES //

var expect = chai.expect,
	assert = chai.assert;


// TESTS //

describe( 'number pdf', function tests() {

	var	validationData = require( './fixtures/accessor.json' ),
		data = validationData.data,
		expected = validationData.expected,
		mu = validationData.mu,
		sigma = validationData.sigma;

	it( 'should export a function', function test() {
		expect( partial ).to.be.a( 'function' );
	});

	it( 'should partially apply the Lognormal pdf for given parameter values', function test() {
		var pdf;
		pdf = partial( mu, sigma);
		expect( pdf ).to.be.a( 'function' );
	});


	it( 'should return a function which evaluates the probability density function', function test() {
		var pdf;
		pdf = partial( mu, sigma);
		for ( var i = 0; i < data.length; i++ ) {
			assert.closeTo( pdf( data[ i ] ), expected[ i ] , 1e-14 );
		}
	});


});

File: test/test.partial.R
options( digits = 16 );
library( jsonlite );

mu = 3
sigma = 6
x = c( 0, 2, 20, 200 )
y = dlnorm( x, mu, sigma )

cat( y, sep = ",\n" )

data = list(
	mu = mu,
	sigma = sigma,
	data = x,
	expected = y
)

write( toJSON( data, digits = 16, auto_unbox = TRUE ), "./test/json/partial.json" )
